kitsune/search/api.py

```diff
@@ -0,0 +1,98 @@
+import itertools
+
+from django.conf import settings
+
+from rest_framework.decorators import api_view
+from rest_framework.response import Response
+
+from kitsune.products.models import Product
+from kitsune.questions.models import QuestionMappingType
+from kitsune.search import es_utils
+from kitsune.sumo.api import GenericAPIException
+from kitsune.wiki.models import DocumentMappingType
+
+
+@api_view(['GET', 'POST'])
```

Comment:
This can read from the body of the text, to support long user input. Although it is possible to have request bodies on GETs, a lot of http libraries don't like it. POST is here simply to allow the nicer use case on dumb libraries. Elasticsearch does the same thing.